### PR TITLE
build dev 1.10 pr17

### PR DESCRIPTION
**Pull request by sdavtaker**
- Preparing generation changes for 1.10
- Force git-dir usage to get the version hash
- Improvements to core-tests
- Type fix for extract_zip declaration on dir_to_delete
- Renamed regenerate script arguments
- Adjustments to code generation

*Issue #1888 *

*Description of changes:*
Updated aspects of generation scripts in both legacy and non legacy builds. 

*Check all that applies:*
- [ ] Did a review by yourself.
- [ ] Added proper tests to cover this PR. (If tests are not applicable, explain.)
- [ ] Checked if this PR is a breaking (APIs have been changed) change.
- [ ] Checked if this PR will _not_ introduce cross-platform inconsistent behavior.
- [ ] Checked if this PR would require a ReadMe/Wiki update.

Check which platforms you have built SDK on to verify the correctness of this PR.
- [ ] Linux
- [ ] Windows
- [ ] Android
- [x] MacOS
- [ ] IOS
- [ ] Other Platforms

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by sbiscigl — APPROVED** [review]

File: tools/scripts/run_code_generation.py [hunk 1]
```diff
@@ -14,6 +14,7 @@
 import zipfile
 from concurrent.futures import ProcessPoolExecutor, wait, FIRST_COMPLETED, ALL_COMPLETED
 from pathlib import Path
+from typing import Optional
```
Comment (SergeyRyabinin) [inline tools/scripts/run_code_generation.py:17]: this file has been updated in main, I'm afraid you need to rebase.

File: tests/aws-cpp-sdk-dynamodb-integration-tests/CMakeLists.txt [hunk 2]
```diff
@@ -13,7 +13,10 @@ if (NOT LEGACY_BUILD)
             ${CMAKE_CURRENT_SOURCE_DIR}/RunTests.cpp
             ${CMAKE_CURRENT_SOURCE_DIR}/TableOperationTest.cpp
     )
-    gtest_discover_tests(aws-cpp-sdk-dynamodb-integration-tests)
+    gtest_discover_tests(
+            aws-cpp-sdk-dynamodb-integration-tests
+            TEST_PREFIX "dynamodb."
```
Comment (SergeyRyabinin) [inline tests/aws-cpp-sdk-dynamodb-integration-tests/CMakeLists.txt:18]: I guess this change shall be repeated for all client tests we have?

File: tests/CMakeLists.txt [hunk 3]
```diff
@@ -1,6 +1,6 @@
 if (BUILD_TESTING)
-    add_subdirectory(../tests/testing-resources)
-    add_subdirectory(../tests/aws-cpp-sdk-core-tests)
+    add_subdirectory(testing-resources)
+    add_subdirectory(aws-cpp-sdk-core-tests)
```
Comment (SergeyRyabinin) [inline tests/CMakeLists.txt:3]: what about the rest?
I would suggest checking if bla-bla-bla-tests exists then add (to cover cases of missing or possible future or manually removed tests (or git sparse checkout).

File: src/aws-cpp-sdk-core/CMakeLists.txt [hunk 4]
```diff
@@ -20,6 +20,7 @@ if (NOT LEGACY_BUILD)
             ${CJSON_LIBRARIES}
             tinyxml2::tinyxml2
             CURL::libcurl
+            OpenSSL::Crypto
```
Comment (SergeyRyabinin) [inline src/aws-cpp-sdk-core/CMakeLists.txt:23]: not non-OpenSSL-build friendly

File: tests/aws-cpp-sdk-dynamodb-integration-tests/CMakeLists.txt (hunk 2 above)
Comment (sdavtaker) [reply tests/aws-cpp-sdk-dynamodb-integration-tests/CMakeLists.txt:18]: Using DynamoDB as reference for the process, once DDB is 100% good and generated, I will expand to others.

File: tools/scripts/run_code_generation.py (hunk 1 above)
Comment (sdavtaker) [reply tools/scripts/run_code_generation.py:17]: I will do it after this merge, I need to rebase over 20 versions.

**Pull request merged**